src/drivers/driver.ts

```diff
@@ -1623,21 +1623,29 @@ export abstract class CMakeDriver implements vscode.Disposable {
             const buildArgs: string[] = this.config.buildArgs.slice(0);
             const buildToolArgs: string[] = ['--'].concat(this.config.buildToolArgs);
 
-            // Only add '-j' argument if parallelJobs > 1
-            if (this.config.numJobs > 1) {
+            const configurationScope = this.workspaceFolder ? vscode.Uri.file(this.workspaceFolder) : null;
+            const parallelJobsSetting = vscode.workspace.getConfiguration("cmake", configurationScope).inspect<number|undefined>('parallelJobs');
+            let numJobs: number | undefined = (parallelJobsSetting?.globalValue || parallelJobsSetting?.workspaceValue || parallelJobsSetting?.workspaceFolderValue);
```

Comment:
This should be done in `config.ts` instead so that any place in the code that needs the setting can have the correct value.